From 3e20223710a1698b166df14b3a95011c72099cc7 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Wed, 9 Jan 2013 17:10:57 +0100
Subject: [PATCH] bootchart: check return of fopen before setvbuf in log.c
 (cherry picked from commit 0908dd2fd5536cf15d75780980ac2eca37ff800f)

---
 src/bootchart/log.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/bootchart/log.c b/src/bootchart/log.c
index 89c7b35..6c40913 100644
--- a/src/bootchart/log.c
+++ b/src/bootchart/log.c
@@ -364,9 +364,9 @@ schedstat_next:
 		if (!ps->smaps) {
 			sprintf(filename, "/proc/%d/smaps", pid);
 			ps->smaps = fopen(filename, "r");
-			setvbuf(ps->smaps, smaps_buf, _IOFBF, sizeof(smaps_buf));
 			if (!ps->smaps)
 				continue;
+			setvbuf(ps->smaps, smaps_buf, _IOFBF, sizeof(smaps_buf));
 		} else {
 			rewind(ps->smaps);
 		}
